test/systemtest/earlgrey/test_sim_verilator.py

```diff
@@ -196,6 +211,24 @@ def app_selfchecking(request, bin_dir):
 # Therefore, if we do not read quickly enough, we miss the PASS/FAIL indication
 # and the test never finishes.
 
+def assert_selfchecking_test_passes(sim):
+    assert sim.find_in_output(
+        re.compile(r"SW test status changed: SwTestStatusInTest$"),
```

Comment:
This check is probably not needed.

Reply:
Even though this check is not strictly needed, it helps to catch errors with broken boot rom images earlier by asserting on an intermediate state that we expect to be present. It matches current behavior, where we did check for the bootrom output.

Reply:
The better approach is to ensure within `sw_test_status_if` that it has transitioned to the SwTestStatusInTest state before transitioning to pass / fail state. I will make a PR for it.